wayland-backend/src/sys/client_impl/mod.rs

```diff
@@ -648,7 +624,7 @@ impl InnerBackend {
             }
         }
 
-        let ret = if guard.evq.is_null() || child_spec.is_none() {
+        let ret = if child_spec.is_none() {
             unsafe {
                 ffi_dispatch!(
                     wayland_client_handle(),
```

Comment:
While this change is technically correct, given the new semantics it would be better to change the "if" branch to invoke `wl_proxy_marshal_array()` instead of `wl_proxy_marshal_array_constructor_versioned()`, as this branch no longer creates new objects ever. That would make this code less confusing to read.

Switching to `wl_proxy_marshal_array()` also requires removing the last two arguments of the call.

Reply:
makes sense, thanks! changed to use `wl_proxy_marshal_array` for the child_spec none branch.